### ISSUE TITLE
Add missing `const` in function signatures

### ISSUE THREAD

**Issue by mpg** [Mbed-TLS/mbedtls#7600]
Add missing `const` in function signatures. See comments below for an incremental list.

**Comment by mpg** [comment]
If still public: `mbedtls_ecp_write_key()`'s first parameter - it should always have been `const`, probably an oversight.

**Comment by gilles-peskine-arm** [comment]
From what I remember from the discussions leading to 3.0, a lot of functions operating on ECP objects have to use a non-const pointer due to the lazy loading of group data, which means that `mbedtls_ecp_group` members have to be non-const in many low-level functions, which then infects the whole interface with non-constness.

**Comment by mpg** [comment]
Yes, but that's only for functions that may end up callling `mbedtls_ecp_mul()`. That's not the case here. (From reading the code, and also trying to add a `const` and see if the compiler complains - it does not.)

**Comment by mpg** [comment]
Btw, I just checked, and `mbedtls_ecp_mul()` is still not quite ready yet to have its first argument `const`, though that would only be a moderate amount of work now that the pre-computed multiples table are stored as static data, instead of computed at runtime. The trouble is, the code still supports computing it at runtime, which is only useful when adding a new curve, as we compute the table at runtime then dump it before we add it to `ecp_curves.c`, and then it never needs to be computed at runtime again - see `scripts/ecp_comb_table.py`.

We should really get rid of that and instead compute the table for new curves in python or using a dedicated C program, but it's ridiculous to keep code in all builds of the library that's only useful when developing the library, not when using it. Especially when it shows in the API, as some parameters that should morally be `const` are not.

So, I'm not adding `ecp_mul()` & Co here. This issue is only for `const` qualifiers that can just be added because they were just forgotten.

**Comment by mpg** [comment]
Created https://github.com/Mbed-TLS/mbedtls/issues/7601 to track preliminary work that's needed before making `ecp_mul()` public.
